integrated_channels/sap_success_factors/client.py

```diff
@@ -31,28 +33,119 @@ def get_oauth_access_token(url_base, client_id, client_secret, company_id, user_
+
+        self.enterprise_configuration = enterprise_configuration
+        self._create_session()
+        super(SAPSuccessFactorsAPIClient, self).__init__()
+
+    def _create_session(self):
+        """
+        Instantiate a new session object for use in connecting with SAP SuccessFactors
+        """
+        session = requests.Session()
+        session.timeout = self.SESSION_TIMEOUT
+
+        oauth_access_token, expires_at = SAPSuccessFactorsAPIClient.get_oauth_access_token(
+            self.enterprise_configuration.sapsf_base_url,
+            self.enterprise_configuration.key,
+            self.enterprise_configuration.secret,
+            self.enterprise_configuration.sapsf_company_id,
+            self.enterprise_configuration.sapsf_user_id
+        )
+
+        session.headers['Authorization'] = 'Bearer {}'.format(oauth_access_token)
+        self.session = session
+        self.expires_at = expires_at
+
+    def send_completion_status(self, payload):
+        """
+        Send a completion status payload to the SuccessFactors OCN Completion Status endpoint
+
+        Args:
+            payload: JSON object (serialized from LearnerDataTransmissionAudit)
+                containing completion status fields per SuccessFactors documentation.
+
+        Returns:
+            The body of the response from SAP SuccessFactors, if successful
+        Raises:
+            HTTPError: if we received a failure response code from SAP SuccessFactors
+        """
+        global_sap_config = SAPSuccessFactorsGlobalConfiguration.current()
+        url = self.enterprise_configuration.sapsf_base_url + global_sap_config.completion_status_api_path
+        return self._call_post(url, payload)
+
+    def send_course_import(self, payload):
+        """
+        Send courses payload to the SuccessFactors OCN Course Import endpoint
+
+        Args:
+            payload: JSON object containing course import data per SuccessFactors documentation.
+
+        Returns:
+            The body of the response from SAP SuccessFactors, if successful
+        Raises:
+            HTTPError: if we received a failure response code from SAP SuccessFactors
+        """
+        global_sap_config = SAPSuccessFactorsGlobalConfiguration.current()
+        url = self.enterprise_configuration.sapsf_base_url + global_sap_config.course_api_path
+        return self._call_post(url, payload)
+
+    def _call_post(self, url, payload):
+        """
+        Make a post request using the session object to a SuccessFactors endpoint.
+
+        Args:
+            url (str): The url to post to.
+            payload: The payload to post.
+        """
+        now = datetime.datetime.utcnow()
+        if now >= self.expires_at:
+            # Create a new session with a valid token
+            self.session.close()
+            self._create_session()
+        response = self.session.post(url, json=payload.serialize())
```

Comment:
If I remember correctly, then the `json` argument takes a Python dict and serializes it to a JSON stream. You want to be using `data`, since the payload is already serialized.